Library/ViewModels/RewardAddOnSelectionViewModel.swift

```diff
@@ -455,3 +529,73 @@ private func isValid(
   return latestSelectedQuantities != selectedRewardQuantities(in: backing)
     || backing.locationId != selectedShippingRule?.location.id
 }
+
+private func checkoutPropertiesData(
```

Comment:
Same here. It's bad practice to copy-paste entire methods/blocks of code.

Reply:
The signatures are the same but the parameters aren't.